daemon/containerd/image.go

```diff
@@ -352,3 +350,132 @@ func imageFamiliarName(img containerdimages.Image) string {
 	}
 	return img.Name
 }
+
+func convertError(err error) error {
+	// TODO: Convert containerd error to Docker error
+	return err
+}
+
+// resolveAllReferences resolves the reference name or ID to an image and returns all the images with
+// the same target.
+func (i *ImageService) resolveAllReferences(ctx context.Context, refOrID string) (*containerdimages.Image, []containerdimages.Image, error) {
```

Comment:
I think it would be worth to highlight that the function can return nil as the first argument (`*containerdimages.Image`) while returning a non empty image slice (`[]containerdimages.Image`) and a nil error. This can only happen if the `refOrID` doesn't match to any image name, only the image target digest?

Reply:
Added a longer section explaining the output. Gofmt seemed to really want me to go big with the formatting but think it is clearer now.